File: PSP/springjavafx/src/main/java/org/example/springjavafx/ui/pantallas/principal/PrincipalController.java
package org.example.springjavafx.ui.pantallas.principal;

import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.fxml.Initializable;
import javafx.scene.layout.BorderPane;
import javafx.scene.layout.Pane;
import org.example.springjavafx.servicios.ServiciosUsuarios;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Component;

import java.io.IOException;

@Component
public class PrincipalController {

    private final ServiciosUsuarios serviciosUsuarios;
    private final ApplicationContext context;
    @FXML
    public BorderPane root;

    public PrincipalController(ServiciosUsuarios serviciosUsuarios, ApplicationContext context) {
        this.serviciosUsuarios = serviciosUsuarios;
        this.context = context;
    }


    public void initialize() {
        System.out.println(serviciosUsuarios.getNombreUsuario());
        cargarPantalla("/fxml/pantalla1.fxml");
    }

    private Pane cargarPantalla(String ruta) {
        Pane panePantalla = null;
        try {

            FXMLLoader fxmlLoader = new FXMLLoader();
            fxmlLoader.setControllerFactory(context::getBean);
            panePantalla = fxmlLoader.load(getClass().getResourceAsStream(ruta));
            root.setCenter(panePantalla);


        } catch (IOException e) {
            System.out.println(e.getMessage());
        }
        return panePantalla;
    }

}

File: PSP/graphql/frontend/src/main/kotlin/org/example/frontend/ui/pantallas/principal/PrincipalController.kt
package org.example.frontend.ui.pantallas.principal

import javafx.fxml.FXML
import javafx.fxml.FXMLLoader
import javafx.fxml.Initializable
import javafx.scene.layout.BorderPane
import javafx.scene.layout.Pane
import javafx.util.Callback
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Component
import java.io.IOException
import java.net.URL
import java.util.*

@Component
class PrincipalController(

    var context: ApplicationContext

) : Initializable {


    @FXML
    lateinit var root: BorderPane





    private fun cargarPantalla(ruta: String): Pane? {
        var panePantalla: Pane? = null
        try {
            val fxmlLoader = FXMLLoader()
            fxmlLoader.setControllerFactory { context.getBean(it) }
            panePantalla = fxmlLoader.load(javaClass.getResourceAsStream(ruta))
            root.center = panePantalla
        } catch (e: IOException) {
            println(e.message)
        }
        return panePantalla
    }


    override fun initialize(location: URL?, resources: ResourceBundle?) {
        cargarPantalla("/fxml/pantalla1.fxml")
    }


}
